### ISSUE TITLE
Support Local Service, Network Service, etc.

### ISSUE THREAD

**Issue by ashic** [mj1856/SimpleImpersonation#8]
It'd be great if we could impersonate built in accounts as well. Perhaps along the lines of:

http://geek.hubkey.com/2008/02/impersonating-built-in-service-account.html

**Comment by mj1856** [comment]
Interesting, but it looks like the trick to doing this is:

 1. Run a console in a service process already under LocalSystem.
 2. Pass the hardcoded strings for the user accounts under the `NT AUTHORTITY` domain.

Sure, I could do the second part, but one could just as easily do that outside of this library and use the current API.

The first part would have to be done outside of this library anyway.  Adding an API for the second part would be misleading.

In other words, I'd not want to include this functionality unless it could work in all cases, not just in the special case of the elevated service process.

Thanks.

**Comment by mattjohnsonpint** [comment]
Just as a follow up, we did indeed add the second part (in #56).  However the first part is still required - one cannot impersonate `LocalSystem` without already running in a process that is already launched with `LocalSystem` as the owner of the process.
